### PR TITLE
Add dim attribute and an example

### PR DESCRIPTION
**Pull request by lausannel**
#81 Add an 'dim' attribute to quiver.Feature class which indicates the number of dimensions
Meanwhile, add an examples of offcial pytorch-geometric and its quiver version, which is now still in progress
Code changes are shown below
```python
#############################
# Original Pyg Code
#############################
# train_loader = NeighborSampler(data.edge_index, sizes=[10, 10], batch_size=256,
#                                shuffle=True, num_nodes=data.num_nodes)
train_loader = torch.utils.data.DataLoader(train_idx,
                                           batch_size=256,
                                           shuffle=True,
                                           drop_last=True)

csr_topo = quiver.CSRTopo(data.edge_index)

quiver_sampler = GraphSageSampler(csr_topo, sizes =[10, 10], device=0)
...
######################
# Original Pyg Code
######################
# for batch_size, n_id, adjs in train_loader:
for seeds in train_loader:
    n_id, batch_size, adjs = sample(data.edge_index, seeds)
```

### CODE REVIEW

File: examples/pyg/graph_sage_unsup_quiver.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+model = SAGE(data.num_node_features, hidden_channels=64, num_layers=2)
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+####################
+# Original Pyg Code
+####################
+x, edge_index = data.x.to(device), data.edge_index.to(device)
```
Comment (eedalong) [inline examples/pyg/graph_sage_unsup_quiver.py:94]: remove `x`, only use `quiver.feature`, but you need to reinstall `torch-scatter` first

File: examples/pyg/graph_sage_unsup_quiver.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+EPS = 1e-15
+
+dataset = 'Cora'
+# path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
+path = '/data/products/' + dataset
+dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
+data = dataset[0]
+
+train_idx = torch.arange(data.num_nodes, dtype=torch.long)
+print(train_idx)
```
Comment (eedalong) [inline examples/pyg/graph_sage_unsup_quiver.py:29]: remove this print

File: srcs/python/quiver/feature.py [hunk 3]
```diff
@@ -212,6 +212,9 @@ def size(self, dim: int):
             shard_tensor = self.clique_tensor_list[clique_id]
             return shard_tensor.size(dim)
 
+    def dim(self):
```
Comment (eedalong) [inline srcs/python/quiver/feature.py:215]: please add function doc

File: examples/pyg/graph_sage_unsup_quiver.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                                           batch_size=256,
+                                           shuffle=True,
+                                           drop_last=True)
+
+csr_topo = quiver.CSRTopo(data.edge_index)
+
+quiver_sampler = GraphSageSampler(csr_topo, sizes =[10, 10], device=0)
+
+def sample(edge_index, batch):
+    # batch = torch.tensor(batch)
```
Comment (eedalong) [inline examples/pyg/graph_sage_unsup_quiver.py:46]: uncomment this line

**Review by eedalong — APPROVED** [review]

File: examples/pyg/graph_sage_unsup_pyg.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+from torch_geometric.nn import SAGEConv
+from torch_geometric.datasets import Planetoid
+from torch_geometric.loader import NeighborSampler as RawNeighborSampler
+
+EPS = 1e-15
+
+dataset = 'Cora'
+# path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
+path = '/data/products/' + dataset
+dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
```
Comment (ZenoTan) [inline examples/pyg/graph_sage_unsup_pyg.py:19]: the path should be sth like `/data/cora`

**Comment by ZenoTan** [comment]
Don't forget to [format](https://github.com/quiver-team/torch-quiver/blob/main/scripts/lint.sh#L44-L45) your code

File: examples/pyg/graph_sage_unsup_quiver.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+import quiver
+from quiver.pyg import GraphSageSampler
+
+EPS = 1e-15
+
+dataset = 'Cora'
+# path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
+path = '/data/products/' + dataset
+dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
+data = dataset[0]
```
Comment (ZenoTan) [inline examples/pyg/graph_sage_unsup_quiver.py:26]: ditto

**Review by ZenoTan — APPROVED** [review]
